File: Spotter.Model/Responses/FriendshipResponse.cs
using Spotter.Model.Enums;

namespace Spotter.Model.Responses
{
    public class FriendshipResponse
    {
        public int Id { get; set; }
        public int RequesterId { get; set; }
        public string RequesterName { get; set; } = string.Empty;
        public int AddresseeId { get; set; }
        public string AddresseeName { get; set; } = string.Empty;
        public FriendshipStatus Status { get; set; }
        public string StatusName { get; set; } = string.Empty;
        public DateTime RequestedAt { get; set; }
        public DateTime? RespondedAt { get; set; }
    }
}

File: UI/spotter_mobile/lib/core/models/friendship_response.dart
class FriendshipResponse {
  final int id;
  final int requesterId;
  final String requesterName;
  final int addresseeId;
  final String addresseeName;
  final int status;
  final String statusName;
  final DateTime requestedAt;
  final DateTime? respondedAt;

  FriendshipResponse({
    required this.id,
    required this.requesterId,
    required this.requesterName,
    required this.addresseeId,
    required this.addresseeName,
    required this.status,
    required this.statusName,
    required this.requestedAt,
    this.respondedAt,
  });

  factory FriendshipResponse.fromJson(Map<String, dynamic> json) {
    return FriendshipResponse(
      id: json['id'] as int,
      requesterId: json['requesterId'] as int,
      requesterName: json['requesterName'] as String,
      addresseeId: json['addresseeId'] as int,
      addresseeName: json['addresseeName'] as String,
      status: json['status'] as int,
      statusName: json['statusName'] as String,
      requestedAt: DateTime.parse(json['requestedAt'] as String),
      respondedAt: json['respondedAt'] != null
          ? DateTime.parse(json['respondedAt'] as String)
          : null,
    );
  }
}
